File: src/codes/Users_Manual/gain/holo_1.cpp
//~#include<autd3.hpp>
//~#include <limits>
//~#include "autd3/gain/holo.hpp"
//~int main() {
const auto backend = std::make_shared<autd3::gain::holo::NalgebraBackend>();
auto g = autd3::gain::holo::GSPAT(backend).with_constraint(
    autd3::gain::holo::EmissionConstraint::Uniform(
        std::numeric_limits<autd3::EmitIntensity>::max()));
//~return 0; }

File: src/codes/Users_Manual/gain/holo_1.py
~from pyautd3 import EmitIntensity
~from pyautd3.gain.holo import GSPAT, EmissionConstraint, NalgebraBackend
~from pyautd3.gain.holo import Pa
~import numpy as np
backend = NalgebraBackend()
~x1 = 0.0
~y1 = 0.0
~z1 = 0.0
~x2 = 0.0
~y2 = 0.0
~z2 = 0.0
~foci = [(np.array([x1, y1, z1]), 5e3 * Pa), (np.array([x2, y2, z2]), 5e3 * Pa)]
g = GSPAT(backend, foci).with_constraint(EmissionConstraint.Uniform(EmitIntensity.maximum()))